### DISCUSSION TITLE
A very tiny bash script to display images

### DISCUSSION THREAD

**Discussion by grondilu** [kovidgoyal/kitty#6574] [Show and tell]
This bash script only requires [ImageMagick](https://imagemagick.org/) (for `convert`), and [GNU Coreutils](https://www.gnu.org/software/coreutils/) (for `base64` and `tr`):

```bash
function icat() {
  convert "${1:-}" png:- |
  base64 -w64 - |
  sed -e '
    $!s/^/m=1;/;
    $ s/^/m=0;/;
    1 s/^/a=T,f=100,t=d,/;
    ' -e "s/^/\x1B_G/; s/$/\x1B\\\\/" |
  tr -d '\n'
}
```

**Comment by kovidgoyal** [comment]
Why -w 64? The chunk size limitation is 4096. And
see here for a script that does not need either tr or sed
https://sw.kovidgoyal.net/kitty/graphics-protocol/#a-minimal-example
to transmit PNG files with BASH

Also note that -w is a GNU extension to base64 its not present on BSD
machines/macOS.

**Reply by grondilu** [reply]
Not sure why I used 64, I think I had to divide 4096 by 64 at some point for some reason. 

I found your bash script a bit convoluted so I wanted something cleaner.

I personally don't mind using GNU extensions, but I understand this is a portability issue.
